src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -526,4 +531,57 @@ bool MPL_gpu_event_is_complete(MPL_gpu_event_t * var)
     return (*var) <= 0;
 }
 
+/* ZE-Specific Functions */
+
+int MPL_ze_init_device_fds(int *num_fds, int *device_fds)
+{
+    const char *device_directory = "/dev/dri/by-path";
+    const char *device_suffix = "-render";
+    struct dirent *ent = NULL;
+    int n = 0;
+
+    DIR *dir = opendir(device_directory);
+    if (dir == NULL) {
+        goto fn_fail;
+    }
+
+    /* Search for all devices in the device directory */
+    while ((ent = readdir(dir)) != NULL) {
+        char dev_name[128];
+
+        if (ent->d_name[0] == '.') {
+            continue;
+        }
+
+        /* They must contain the device suffix */
+        if (strstr(ent->d_name, device_suffix) == NULL) {
+            continue;
+        }
+
+        strcpy(dev_name, device_directory);
+        strcat(dev_name, "/");
+        strcat(dev_name, ent->d_name);
+
+        /* Open the device */
+        if (device_fds) {
+            device_fds[n] = open(dev_name, O_RDWR);
+        }
+
+        n++;
+    }
+
+    *num_fds = n;
+
+  fn_exit:
+    return MPL_SUCCESS;
+  fn_fail:
+    return MPL_ERR_GPU_INTERNAL;
+}
+
+void MPL_ze_set_fds(int num_fds, int *fds)
+{
+    shared_device_fds = fds;
+    shared_device_fd_count = num_fds;
+}
+
```

Comment:
Can both these can be static functions inside `ipc_fd.c`?

Reply:
See previous comment about usage of `shared_device_fds`